Typescript issue with date in where

<!--
If you don't follow the issue template, your issue may be closed.
Please note this is an issue tracker, not a support forum.
For general questions, please use StackOverflow or Slack.
-->

## Issue Description
Date types are not accepted as an option in a findAll where.

### What are you doing?

<!--
Post a MINIMAL, SELF-CONTAINED code that reproduces the issue. It must be runnable by simply copying and pasting into an isolated JS file, except possibly for the database connection configuration.
Check http://sscce.org/ or https://stackoverflow.com/help/minimal-reproducible-example to learn more about SSCCE/MCVE/reprex.
-->

```js
import { Sequelize, Model, DataTypes, BuildOptions } from 'sequelize';

const sequelize = new Sequelize('mysql://root:asd123@localhost:3306/mydb');

// We need to declare an interface for our model that is basically what our class would be
interface MyModel extends Model {
    readonly id: number;
    dt: Date;
  }

  // Need to declare the static model so `findOne` etc. use correct types.
  type MyModelStatic = typeof Model & {
    new (values?: object, options?: BuildOptions): MyModel;
  }

  // TS can't derive a proper class definition from a `.define` call, therefor we need to cast here.
  const MyDefineModel = <MyModelStatic>sequelize.define('MyDefineModel', {
    id: {
      primaryKey: true,
      type: DataTypes.INTEGER.UNSIGNED,
    },
    dt: { type: DataTypes.DATE, allowNull: true },
  });

  function stuffThree() {
    MyDefineModel.findAll({ where: { dt: new Date() } });
  }
```

### What do you expect to happen?

<!-- Explain what behavior you wanted/expected. You may include an output. -->

tsc should compile the file without issue.

### What is actually happening?

<!-- Show what happened. -->

tsc complains that a Date is not assignable to `WhereAttributeHash`
I feel like `WhereValue` should contain Date, but I assume it is not there for a reason?

```
test.ts:26:29 - error TS2322: Type '{ dt: Date; }' is not assignable to type 'WhereOptions'.
  Type '{ dt: Date; }' is not assignable to type 'WhereAttributeHash'.
    Property 'dt' is incompatible with index signature.
      Type 'Date' is not assignable to type 'string | number | boolean | WhereAttributeHash | AndOperator | OrOperator | Literal | Where | Fn | Col | WhereOperators | WhereGeometryOptions | (string | number | WhereAttributeHash)[]'.
        Type 'Date' is not assignable to type 'WhereAttributeHash'.
          Index signature is missing in type 'Date'.

26     MyDefineModel.findAll({ where: { dt: new Date() } });
                               ~~~~~

  node_modules/sequelize/types/lib/model.d.ts:64:3
    64   where?: WhereOptions;
         ~~~~~
    The expected type comes from property 'where' which is declared here on type 'FindOptions'


Found 1 error.
```

### Environment

- Sequelize version: 5.17.2
- Node.js version: v10.15.3
- Operating System: Ubuntu 18.04.3 LTS
- If TypeScript related: TypeScript version: 3.4.5

## Issue Template Checklist

<!-- Please answer the questions below. If you don't, your issue may be closed. -->

### How does this problem relate to dialects?

<!-- Choose one. -->

- [x] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s): <!-- Put dialect(s) here -->
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

<!-- Remember that first contributors are welcome! -->

- [x] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

--- a/types/lib/model.d.ts
+++ b/types/lib/model.d.ts
@@ -341,6 +341,7 @@ export type WhereValue =
   | string // literal value
   | number // literal value
   | boolean // literal value
+  | Date // literal value
   | Buffer // literal value
   | null
   | WhereOperators

--- a/types/test/where.ts
+++ b/types/test/where.ts
@@ -22,6 +22,7 @@ where = {
   buffer: Buffer.alloc(0),
   buffers: [Buffer.alloc(0)],
   null: null,
+  date: new Date()
 };
 
 // Operators

